core/peerset/src/lib.rs

```diff
@@ -646,5 +653,40 @@ mod tests {
 
 		futures::executor::block_on(fut);
 	}
+
+
+	#[test]
+	fn test_peerset_reserved_node_not_banned() {
+		let bootnode = PeerId::random();
+		let reserved_peer = PeerId::random();
+		let config = PeersetConfig {
+			in_peers: 0,
+			out_peers: 1,
+			bootnodes: vec![bootnode],
+			reserved_only: true,
+			reserved_nodes: Vec::new(),
+		};
+
+		let (mut peerset, handle) = Peerset::from_config(config);
+		handle.add_reserved_peer(reserved_peer.clone());
+
+		// Setting reputation of reserved node under the threshold.
+		handle.report_peer(reserved_peer.clone(), BANNED_THRESHOLD - 1);
+
+		let fut = futures::future::poll_fn(move |cx| {
+			// We need one polling for the message to be processed.
+			assert_eq!(Stream::poll_next(Pin::new(&mut peerset), cx), Poll::Pending);
```

Comment:
need to change to (please run formatter, mine is not working after restarting pc lol):
```rust
assert_eq!(Stream::poll_next(Pin::new(&mut peerset), cx), Poll::Ready(Some(Message::Connect(reserved_peer.clone()))));
```